s3torchconnector/src/s3torchconnector/s3reader/constructor.py

```diff
@@ -97,10 +131,11 @@ def get_reader_type_string(
                 S3ReaderConstructor.default()
             )
 
-        if not isinstance(constructor, partial):
+        if isinstance(constructor, DCPOptimizedConstructor):
```

Comment:
Same here - this feels pretty janky to me. What's this used for? Just debugging or to actually do something based on it?